pennylane/_qubit_device.py

```diff
@@ -401,6 +406,129 @@ def execute_new(self, circuit, **kwargs):
         #     self.tracker.record()
         return results
 
+    def shot_vec_statistics(self, circuit):
+        """Process measurement results from circuit execution using a device
+        with a shot vector and return statistics.
+
+        This is an auxiliary method of execute_new and uses statistics_new.
+
+        When using shot vectors, measurement results for each item of the shot
+        vector are contained in a tuple.
+
+        Args:
+            circuit (~.tapes.QuantumTape): circuit to execute on the device
+
+        Raises:
+            QuantumFunctionError: if the value of :attr:`~.Observable.return_type` is not supported
+
+        Returns:
+            tuple: stastics for each shot item from the shot vector
+        """
+        results = []
+        s1 = 0
+
+        ret_types = [m.return_type for m in circuit.measurements]
+        counts_exist = any(ret is qml.measurements.Counts for ret in ret_types)
+        single_measurement = len(circuit.measurements) == 1
+
+        for shot_tuple in self._shot_vector:
+            s2 = s1 + np.prod(shot_tuple)
+            r = self.statistics_new(
+                circuit.observables, shot_range=[s1, s2], bin_size=shot_tuple.shots
+            )
+
+            # This will likely be required:
+            # if qml.math._multi_dispatch(r) == "jax":  # pylint: disable=protected-access
+            #     r = r[0]
+
+            if single_measurement:
+                r = r[0] if counts_exist else qml.math.squeeze(r)
+            else:
+                if shot_tuple.copies == 1:
+                    r = tuple(
+                        r_[0]
+                        if isinstance(r_, list)  # need to unwrap the single element
+                        else qml.math.squeeze(r_).T
+                        for idx, r_ in enumerate(r)
+                    )
+                else:
+
+                    if counts_exist:
+                        r = self._multi_meas_with_counts_shot_vec(circuit, shot_tuple, r)
+                    else:
+                        # r is a nested sequence, contains the results for multiple measurements
+                        # Each item of r has copies length
+                        r = [
+                            tuple(self._asarray(r_.T[idx]) for r_ in r)
```

Comment:
Is the `asarray` needed because of the transpose? What is the purpose of transposing?

Reply:
Left comments in https://github.com/PennyLaneAI/pennylane/pull/2815/commits/992e1ae303bab2b8a61075fd8722690b3fcc6191. These processing steps are due to how internal methods like `probs` and `sample` provide results.

Reply:
Thank you for the clear explanation!